src/nvim/tui/tui.c

```diff
@@ -1021,10 +1053,11 @@ static void fix_terminfo(TUIData *data)
 end:
   // Fill some empty slots with common terminal strings
   if (data->term == kTermiTerm) {
-    data->unibi_ext.set_cursor_color = (int)unibi_add_ext_str(
+    data->unibi_ext.set_cursor_bg_color = (int)unibi_add_ext_str(
         ut, NULL, "\033]Pl%p1%06x\033\\");
   } else {
-    data->unibi_ext.set_cursor_color = (int)unibi_add_ext_str(
+    // new form rgb:X/X/X does not seem to work so well
```

Comment:
vte is [supposed](https://git.gnome.org/browse/vte/tree/src/vtetypes.cc?id=a31a41d2a93ef45c3940d714f82f58c6b3f9d7d8#n84) to convert the `rgb:rrr/ggg/bbb` (independent of the arity of each color) to the `#rrrgggbbb` form and then pass that on to [`pango_color_parse`](https://developer.gnome.org/pango/stable/pango-Text-Attributes.html#pango-color-parse).

Older versions of vte used [`gdk_parse_color`](https://developer.gnome.org/gdk3/stable/gdk3-Colors.html#gdk-color-parse) instead, but either one should handle the format fine.

If you're seeing different behavior, then I'd either guess it's an old version of vte or the terminal using vte is doing something wrong.